Align forbidden host and domain code points with spec and other browsers

achristensen07 opened the pull request:
#### 235db6f25f6187f42e36d21ca7b4559e20c14dd6
<pre>
Align forbidden host and domain code points with spec and other browsers
<a href="https://bugs.webkit.org/show_bug.cgi?id=257213">https://bugs.webkit.org/show_bug.cgi?id=257213</a>
rdar://109725285

Reviewed by Chris Dumez.

<a href="https://url.spec.whatwg.org/#forbidden-host-code-point">https://url.spec.whatwg.org/#forbidden-host-code-point</a> and <a href="https://url.spec.whatwg.org/#forbidden-domain-code-point">https://url.spec.whatwg.org/#forbidden-domain-code-point</a>
are different sets of code points, the latter used when parsing special URL schemes and the former used when parsing
non-special URL schemes, though there is a large overlap.  Chrome and Firefox pass the tests that this PR makes WebKit pass.

* LayoutTests/imported/w3c/web-platform-tests/url/a-element-xhtml_exclude=(file_javascript_mailto)-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/url/a-element_exclude=(file_javascript_mailto)-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/url/url-constructor.any.worker_exclude=(file_javascript_mailto)-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/url/url-constructor.any_exclude=(file_javascript_mailto)-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/url/url-origin.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/url/url-origin.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/url/url-setters-stripping.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/url/url-setters-stripping.any.worker-expected.txt:
* Source/WTF/wtf/URLParser.cpp:
(WTF::isC0Control):
(WTF::URLParser::isForbiddenHostCodePoint):
(WTF::URLParser::isForbiddenDomainCodePoint):
(WTF::URLParser::hasForbiddenHostCodePoint):
(WTF::URLParser::parseHostAndPort):
(WTF::isForbiddenHostCodePoint): Deleted.
* Source/WTF/wtf/URLParser.h:

Canonical link: <a href="https://commits.webkit.org/264482@main">https://commits.webkit.org/264482@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/afd695a0272693ba389d71848ce4caf2500c7a2a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/7604 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/7872 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/8056 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/9247 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/7793 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/9909 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/7799 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/10665 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/7737 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/8971 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/7013 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/9383 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/6225 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/6925 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/14645 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/6493 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/7388 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/7047 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/10561 "8 api tests failed or timed out") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/7195 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/7542 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/6158 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/7759 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/6880 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/1773 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/11091 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/7964 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/931 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/7273 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/1916 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

cdumez reviewed (approved):
LGTM if the bots are happy. Thanks for fixing.

achristensen07 committed:
Align forbidden host and domain code points with spec and other browsers
https://bugs.webkit.org/show_bug.cgi?id=257213
rdar://109725285

Reviewed by Chris Dumez.

https://url.spec.whatwg.org/#forbidden-host-code-point and https://url.spec.whatwg.org/#forbidden-domain-code-point
are different sets of code points, the latter used when parsing special URL schemes and the former used when parsing
non-special URL schemes, though there is a large overlap.  Chrome and Firefox pass the tests that this PR makes WebKit pass.

* LayoutTests/imported/w3c/web-platform-tests/url/a-element-xhtml_exclude=(file_javascript_mailto)-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/url/a-element_exclude=(file_javascript_mailto)-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/url/url-constructor.any.worker_exclude=(file_javascript_mailto)-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/url/url-constructor.any_exclude=(file_javascript_mailto)-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/url/url-origin.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/url/url-origin.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/url/url-setters-stripping.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/url/url-setters-stripping.any.worker-expected.txt:
* Source/WTF/wtf/URLParser.cpp:
(WTF::isC0Control):
(WTF::URLParser::isForbiddenHostCodePoint):
(WTF::URLParser::isForbiddenDomainCodePoint):
(WTF::URLParser::hasForbiddenHostCodePoint):
(WTF::URLParser::parseHostAndPort):
(WTF::isForbiddenHostCodePoint): Deleted.
* Source/WTF/wtf/URLParser.h:

Canonical link: https://commits.webkit.org/264482@main
